bin/mixlib-install

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env ruby
+
+require "mixlib/install"
+require "mixlib/cli"
+require "mixlib/shellout"
+
+module Mixlib
+  class Install
+    class Download
+      include Mixlib::CLI
+
+      option :dir,
+      short: "-d DIRECTORY",
+      long: "--dir DIRECTORY",
+      description: "Directory to download the file. Defaults to current directory",
+      default: Dir.pwd
+
+      option :product,
+      short: "-p PRODUCT",
+      long: "--product PRODUCT",
+      description: "The product to download",
+      required: true
+
+      option :channel,
+      short: "-c CHANNEL",
+      long: "--channel CHANNEL",
+      description: "The channel from which to download the specified product. Defaults to stable",
+      default: :stable
+
+      option :version,
+      short: "-v VERSION",
+      long: "--version VERSION",
+      description: "The version of the product to download. Defaults to latest",
+      default: :latest
+
+      option :platform,
+      long: "--platform PLATFORM",
+      description: "The platform which the product will be installed. If no platform is provided it will use the current platform."
+
+      option :platform_version,
+      long: "--platform-version PLATFORM_VERSION",
+      description: "The version of the platform on which the product will be installed"
+
+      option :architecture,
+      long: "--arch ARCHITECTURE",
+      description: "The architecture of the platform"
+    end
+  end
+end
+
+cli = Mixlib::Install::Download.new
+cli.parse_options
+
+# Set mininum options
+options = {
+  channel: cli.config[:channel].to_sym,
+  product_name: cli.config[:product],
+  product_version: cli.config[:version],
+}
+
+# Set platform info or auto detect platform
+if cli.config[:platform]
+  if cli.config[:platform_version].nil? || cli.config[:architecture].nil?
+    abort "Must provide platform version and architecture when specifying a platform"
+  end
+  options[:platform] = cli.config[:platform]
+  options[:platform_version] = cli.config[:platform_version]
+  options[:architecture] = cli.config[:architecture]
+else
+  options.merge!(Mixlib::Install.detect_platform)
+end
+
+puts "Querying for product with options: #{options}"
+artifact = Mixlib::Install.new(options).artifact_info
+abort "No results found with options: #{options}" if artifact.nil? || artifact.is_a?(Array)
+
+file = File.join(cli.config[:dir], File.basename(artifact.url))
+puts "Starting download #{artifact.url} to #{file}"
+command = "wget --progress=bar:force #{artifact.url} -O #{file}"
+
+cmd = Mixlib::ShellOut.new(command)
+cmd.live_stream ||= $stdout.tty? ? $stdout : nil
+cmd.run_command
+
+puts "Download complete"
+
+exit 0
```

Comment:
It would be nice to get some unit and functional tests on this CLI. Maybe moving the core logic into subcommand classes will make that testing easier.

The functional test for Omnibus's net fetcher might give you some ideas:
https://github.com/chef/omnibus/blob/master/spec/functional/fetchers/net_fetcher_spec.rb